source/cdk/lib/msam-dynamodb.ts

```diff
@@ -0,0 +1,340 @@
+/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+       SPDX-License-Identifier: Apache-2.0 */
+
+import {
+    aws_dynamodb as dynamodb,
+    aws_lambda as lambda,
+    aws_iam as iam,
+    aws_s3 as s3,
+    Aws,
+    CustomResource,
+    CustomResourceProps,
+    Duration,
+    Fn,
+    NestedStack,
+    NestedStackProps,
+    RemovalPolicy,
+} from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as utils from './utils';
+
+interface MsamDynamoDBProps extends NestedStackProps {
+    /**
+     * This is the IAM Role ARN for the DynamoDB Lambda functions.
+     */
+    readonly DynamoDBIAMRoleARN: string;
+}
+
+export class MsamDynamoDB extends NestedStack {
+    /**
+     * Alarms table resource name
+     */
+    readonly AlarmsTable: string;
+
+    /**
+     * Channels table resource name
+     */
+    readonly ChannelsTable: string;
+
+    /**
+     * Content table resource name
+     */
+    readonly ContentTable: string;
+
+    /**
+     * Events table resource name
+     */
+    readonly EventsTable: string;
+
+    /**
+     * Layout table resource name
+     */
+    readonly LayoutTable: string;
+
+    /**
+     * Settings table resource name
+     */
+    readonly SettingsTable: string;
+
+    /**
+     * CloudWatchEvents table resource name
+     */
+    readonly CloudWatchEventsTable: string;
+
+    /**
+     * Notes table resource name
+     */
+    readonly NotesTable: string;
+
+    constructor(scope: Construct, id: string, props: MsamDynamoDBProps) {
+        super(
+            scope,
+            id,
+            {
+                ...props,
+                description: 'Media Services Application Mapper DynamoDB Tables %%VERSION%%',
+            }
+        );
+
+        /**
+         * Lambda Functions
+         */
+        const defaultSettingsResourceFunction = new lambda.Function(this, 'DefaultSettingsResource', {
+            code: lambda.Code.fromBucket(
+                s3.Bucket.fromBucketName(
+                    this,
+                    'DistributionBucket',
+                    Fn.join('-', ['%%BUCKET_NAME%%', Aws.REGION]),
+                ),
+                '%%SOLUTION_NAME%%/%%VERSION%%/dynamodb_resource_DEV_0_0_0.zip',
+            ),
+            handler: 'lambda_function.lambda_handler',
+            description: 'MSAM Lambda for adding defaults to the Settings table',
+            memorySize: 2560,
+            environment: {
+                BUILD_STAMP: 'DEV_0_0_0',
+                SOLUTION_ID: 'AwsSolution/SO0048/%%VERSION%%',
+            },
+            role: iam.Role.fromRoleArn(this, 'DynamoDBIAMRole', props.DynamoDBIAMRoleARN),
```

Comment:
I would suggest a slightly simpler approach....

Instead of distilling the role down to it's ARN to share with the nested stack and then reconstructing the role from the ARN, I would suggest passing in the role object as the nested stack parameter/argument and then use it directly here.